cli/src/datafeed.ts

```diff
@@ -195,6 +224,168 @@ export function bulkRemoveHandler() {
   return wrapper
 }
 
+export function bulkActivateHandler() {
+  async function wrapper({ data }: { data }) {
+    const bulkData = data as IDatafeedBulk
+
+    const chain = bulkData?.chain || 'localhost'
+    const service = bulkData?.service || 'DATA_FEED'
+
+    const fetcherHost = bulkData?.fetcherHost || 'http://fetcher.orakl.svc.cluster.local'
+    const workerHost = bulkData?.workerHost || 'http://aggregator-worker.orakl.svc.cluster.local'
+    const listenerHost =
+      bulkData?.listenerHost || 'http://aggregator-listener.orakl.svc.cluster.local'
+    const reporterHost =
+      bulkData?.reporterHost || 'http://aggregator-reporter.orakl.svc.cluster.local'
+
+    const fetcherPort = bulkData?.fetcherPort || '4040'
+    const workerPort = bulkData?.workerPort || '5000'
+    const listenerPort = bulkData?.listenerPort || '4000'
+    const reporterPort = bulkData?.reporterPort || '6000'
+
+    const listeners = await listenerListHandler()({ chain, service })
+    const reporters = await reporterListHandler()({ chain, service })
+
+    if (!checkBulkSource(data?.bulk)) {
+      console.error(`invalid json src format`)
+      return
+    }
+
+    for (const activateElement of data.bulk) {
+      const aggregatorData = await loadJsonFromUrl(activateElement.aggregatorSource)
+      if (!checkAggregatorSource(aggregatorData)) {
+        console.error(`invalid aggregatorData: ${aggregatorData}, skipping activation`)
+        continue
+      }
+
+      const reporterId = reporters.find(
+        (reporter) =>
+          reporter.address.toLowerCase() == activateElement.reporter.walletAddress.toLowerCase()
+      )?.id
+      if (!reporterId) {
+        console.error(
+          `reporterId not found for ${activateElement.reporter.walletAddress}, skipping activation`
+        )
+        continue
+      }
+
+      const listenerId = listeners.find(
+        (listener) => listener.address == aggregatorData.address
+      )?.id
+      if (!listenerId) {
+        console.error(`listenerId not found for ${aggregatorData.address}, skipping activation`)
+        continue
+      }
+
+      await fetcherStartHandler()({
```

Comment:
@nick-bisonai I think we should consider to have circuit breaker when things do not go as planned. It is possible that something is not going to get activated, then in my opinion should stop the process of activation. What do you think? Currently, the handlers used in this function do not return error values, but I think we should consider to add them and prevent from further activation is some of them throws an error.

Reply:
@martinkersner I totally agree. Can I make it an issue and leave if for future updates?

Reply:
Sure! Let's make an issue for it ;)